### PR TITLE
Issue 3385 cpu usage android

### LINKED ISSUE #3385: Need alternative way to get CPU usage on Android 8+ devices (DEF-3497)
The `profiler.get_cpu_usage()` function is not working on Android 8 and above. Since Android 8 apps don't have permission to read `/proc/stat`. We need to figure out an alternative way to get the CPU usage for newer Android devices.

### PR DESCRIPTION
**Pull request by otsakir**
Alternative CPU usage algorithm for android 8+. Fixes profiler.get_cpu_usage report as 0 on late android versions.

fixes #3385

### CODE REVIEW

**Comment by britzl** [comment]
@otsakir don't forget the checklist! 👇 

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

**Comment by otsakir** [comment]
Thanks @britzl . I checked only applicable list items.

**Comment by britzl** [comment]
> I checked only applicable list items.

Thanks. I also assigned the issue itself to project 1.4.2 since this fix will be included in that release.

**Review by JCash — DISMISSED** [review]
Some smaller code changes, but otherwise looks good.

File: engine/profiler/src/profiler_proc_utils.cpp [hunk 1]
```diff
@@ -76,7 +77,59 @@ static long int ParseProcStat()
...
+    int max_cpuid = -1;
+
+    // go for "processor : *" lines and keep the processor id.
+    while ( result != EOF)
+    {
+            result = fscanf(fp, "processor : %d \n", &cpuid);
+            if (result == EOF)
+            {
+                    break;
+            } else
```
[the remark this answers is not in the document]
Comment (otsakir) [reply engine/profiler/src/profiler_proc_utils.cpp:100]: ok. I just removed the "else".

**Review by JCash — APPROVED** [review]

**Pull request merged**